app/models/topic.rb

```diff
@@ -47,6 +47,10 @@ class Topic < ActiveRecord::Base
 
   acts_as_readable
 
+  def self.recent_topics
+    Post.order('id desc').select('DISTINCT topic_id').limit(5).map(&:topic)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.